warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt

```diff
@@ -1,281 +1,653 @@
-import androidx.compose.material3.Icon
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.BlendMode
+import androidx.compose.ui.graphics.ColorFilter
 import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.graphics.vector.rememberVectorPainter
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.MeasurePolicy
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.IntOffset
 import androidx.compose.ui.unit.dp
-import androidx.constraintlayout.compose.ChainStyle
-import androidx.constraintlayout.compose.ConstrainedLayoutReference
-import androidx.constraintlayout.compose.ConstraintLayout
-import androidx.constraintlayout.compose.Dimension
+import com.schibsted.nmp.warp.theme.LocalColors
+import com.schibsted.nmp.warp.theme.LocalDimensions
+import com.schibsted.nmp.warp.theme.LocalTypography
+import com.schibsted.nmp.warp.theme.StepIndicatorDimensions
 import com.schibsted.nmp.warp.theme.WarpBrandedTheme
-import com.schibsted.nmp.warp.theme.WarpTheme
+import com.schibsted.nmp.warp.theme.WarpTheme.colors
+import com.schibsted.nmp.warp.theme.WarpTheme.dimensions
+import com.schibsted.nmp.warp.theme.WarpTheme.typography
 import com.schibsted.nmp.warp.utils.FlavorPreviewProvider
+import java.lang.Float.min
+import kotlin.math.floor
+import kotlin.math.max
+
 
 /**
- * A component to show a user where they are in a user journey
+ * A vertical component to show a user where they are in a user journey
  * For more info [see](https://warp-ds.github.io/tech-docs/components/steps/)
  * @param modifier Modifier applied to the component
- * @param steps Amount of steps
+ * @param steps Amount of steps. Must be greater than 1
  * @param activeStep Currently active step
- * @param orientation Orientation to draw the steps in
  * @param onStepClicked will be invoked if a step is clicked
- * @param stepTitle Optional composable that will be drawn with the step
- * @param stepDescription Optional composable that will be drawn below the title
+ * @param stepTitle Function to provide titles for the steps
+ * @param stepContent Composable containing the content of each step
  */
 @Composable
```

Comment:
We need to support xml aswell, so we need to add a AbstractComposeView class of this component

Reply:
Yeah I know, this PR only adds animations though so I would say it is out of scope for this PR